lms/djangoapps/grades/models.py

```diff
@@ -575,6 +575,24 @@ def prefetch(cls, course_id, users):
             cls.objects.filter(user_id__in=[user.id for user in users], course_id=course_id)
         }
 
+    @classmethod
+    def prefetch_by_user(cls, user, course_keys):
+        """
+        Prefetches grades for the given user for each given course
+
+        Note that this may clobber data from prefetches done by a different access pattern
+        Interleave different means of prefetching at your peril
+        """
+        cache = get_cache(cls._CACHE_NAMESPACE)
+        grades = cls.objects.filter(user_id=user.id)
+        # we don't want to generate huge WHERE course_id IN (...) clauses in SQL
+        if len(course_keys) < 100:
+            grades = grades.filter(course_id__in=course_keys)
+        for grade in grades:
+            cache[grade.course_id] = {
```

Comment:
I think you need to key by `cls._cache_key(grade.course_id)` here.

Reply:
there's no rollsafe github emoji and it's really cramping my groove

but thankfully we may be taking a more fire and brimstone approach to this instead